Tools/WebKitTestRunner/TestController.cpp

```diff
@@ -360,8 +361,28 @@ void TestController::setIsMediaKeySystemPermissionGranted(bool granted)
     m_isMediaKeySystemPermissionGranted = granted;
 }
 
-static void queryPermission(WKStringRef, WKSecurityOriginRef, WKQueryPermissionResultCallbackRef callback)
+static void queryPermission(WKStringRef string, WKSecurityOriginRef securityOrigin, WKQueryPermissionResultCallbackRef callback)
 {
+    TestController::singleton().handleQueryPermission(string, securityOrigin, callback);
+}
+
+void TestController::handleQueryPermission(WKStringRef string, WKSecurityOriginRef securityOrigin, WKQueryPermissionResultCallbackRef callback)
```

Comment:
handQueryPermission was added because TestController::queryPermission needs to access a private data member of TestController, but it is a static method, so it can't access it. One fix was to add a function like handleQueryPermission and the other was to add a getter function to return the data member. Adding handleQueryPermission seems to be the preferred method.

Reply:
For updating Notification::permission() to have the same behavior as Permissions::query() to reduce fingerprinting--you're right. The Permissions API feature is enabled, so we need to ensure that these two functions continue to have the same behavior. I've updated Notification::permission() in a similar way to Permissions::query().